### PR TITLE
Keyboard Hook

### PR DESCRIPTION
**Pull request by cookta2012**
This is the beginning of a low level keyboard hook

### CODE REVIEW

**Review by cookta2012 — COMMENTED** [review]
@Ylianst

File: KVMControlHook.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        keyStatus = 0;
+                        break;
+                    case WM_KEYUP:
+                        keyStatus = 1;
+                        break;
+                    case WM_SYSKEYDOWN:
+                        keyStatus = 4;
+                        break;
+                    case WM_SYSKEYUP:
+                        keyStatus = 5;
```
Comment (cookta2012) [inline KVMControlHook.cs:78]: I should be able to pass them as a standard down and up... Hmm..

**Comment by Ylianst** [comment]
Very interesting! It says it's a draft pull request so I can't merge it. But very interested by this. I can't wait to see the Windows key work correctly.

**Comment by Ylianst** [comment]
Ok, got it merged and it works wonderfully! That is amazing.

**Comment by cookta2012** [comment]
No problem

**Comment by cookta2012** [comment]
My only concern is I was sending the keys correctly for your sendKey function

**Comment by cookta2012** [comment]
Please review KVMControlHook.cs line 68-81 regarding the keystate which I see in your code is a flag

**Comment by Ylianst** [comment]
Ok, checking this now.

**Comment by Ylianst** [comment]
So, I am using a mix of both systems for now. It seems to work well. Let me know if this is ok for you.

**Comment by cookta2012** [comment]
That is fine

**Comment by cookta2012** [comment]
My curiosity is piqued as to why you prefer the method you have over unifying it all to a single keyboard hook. With that decision it would impact the ability to implement a way of inputting input to multiple clients.

**Comment by cookta2012** [comment]
My curiosity is piqued as to why you prefer the method you have over unifying it all to a single keyboard hook. With that decision it would impact the ability to implement a way of inputting input to multiple clients.

**Comment by Ylianst** [comment]
This is just temporary, I need to do more testing on this. A single keyboard hook is certainly simpler.

**Comment by cookta2012** [comment]
On line 54 I had that split up, as a debugging measure. Sometimes I was getting could not cast a disposed object, or could not marshal a disposed object occasionally, but when I had it split it wasn't doing that dispose bit incorrectly

**Pull request merged**